cpp/src/filling/repeat.cu

```diff
@@ -48,22 +76,26 @@ gdf_column repeat(const gdf_column &in, const gdf_column& count, cudaStream_t st
   
   thrust::inclusive_scan(exec_policy, count_data, count_data + count.size, offset.begin());
 
-  gdf_size_type output_size = offset.back();
-
-  rmm::device_vector<gdf_size_type> indices(output_size);
-  thrust::upper_bound(exec_policy,
-                      offset.begin(), offset.end(),
-                      thrust::make_counting_iterator(0),
-                      thrust::make_counting_iterator(output_size),
-                      indices.begin());
+  return fill_between_offsets(in, offset, stream);
+}
 
-  gdf_column output = cudf::allocate_like(in, output_size);
+gdf_column repeat(const gdf_column &in, const gdf_scalar& count, cudaStream_t stream = 0) {
+  CUDF_EXPECTS(count.dtype == gdf_dtype_of<gdf_size_type>(),
+    "Count value should be of index type");
+  CUDF_EXPECTS(count.is_valid, "count cannot be null");
 
-  cudf::table in_table{const_cast<gdf_column*>(&in)};
-  cudf::table out_table{&output};
-  cudf::gather(&in_table, indices.data().get(), &out_table);
+  if (in.size == 0) {
+    return cudf::empty_like(in);
+  }
+  
+  rmm::device_vector<gdf_size_type> offset(in.size);
+  
+  thrust::inclusive_scan(rmm::exec_policy(stream)->on(stream),
```

Comment:
Do you need a scan at all in this case? Can you just use a counting_iterator with a stride? If that doesn't exist, create a transform_iterator from a counting iterator with a lambda that multiplies its input by the count.

```
int stride = count.data.si32;
auto strider = [stride](int i) { return i * stride; };
auto begin = thrust::make_transform_iterator(thrust::make_counting_iterator(0), strider);
auto end = thrust::make_transform_iterator(thrust::make_counting_iterator(in.size), strider);
```

Then modify `fill_between_offsets` take a begin and end iterator rather than a device vector.

Reply:
Hmmm, I realized this is not right.